### PR TITLE
Try b2g gsi

### PR DESCRIPTION
**Pull request by ittat**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,21 @@ At boot time, you might need `adb shell setenforce 0` for B2G to boot (flash_sar
   
 If need to output a zip ROM file, you can use `./build.sh dist DIST_DIR=dist_output` instead of `./build.sh` in step 5.
 
+## Generic B2G System Images (GSI)
+
+1. Fetch the code: `REPO_INIT_FLAGS="--depth=1" ./config.sh gsi_b2g`
```
Comment (lissyx) [inline README.md:69]: needs `GITREPO=https://github.com/ittat/manifests BRANCH=b2g-gsi` for testing

File: README.md [hunk 2]
```diff
@@ -64,6 +64,21 @@ At boot time, you might need `adb shell setenforce 0` for B2G to boot (flash_sar
...
+3. Install Gecko dependencies: `cd gecko && ./mach bootstrap`, choose option 4 (Android Geckoview).
+4. Apply patch: `./patcher/patcher.sh`
+5. Build: `./build-gsi.sh [type name] systemimage`
+
+   type:
+   - `gsi_arm64_ab` (arm64 partitionA/B)
+   - `gsi_arm64_a` (arm64 partitionA)
+   - `gsi_arm_ab` (arm partitionA/B)
+   - `gsi_arm_a` (arm partitionA)
+6. Flash: [click it](https://source.android.google.com/setup/build/gsi#flashing-gsis)
```
Comment (lissyx) [inline README.md:80]: https://source.android.com/setup/build/gsi#flashing-gsis

File: README.md [hunk 3]
```diff
@@ -64,6 +64,21 @@ At boot time, you might need `adb shell setenforce 0` for B2G to boot (flash_sar
   
 If need to output a zip ROM file, you can use `./build.sh dist DIST_DIR=dist_output` instead of `./build.sh` in step 5.
 
+## Generic B2G System Images (GSI)
+
+1. Fetch the code: `REPO_INIT_FLAGS="--depth=1" ./config.sh gsi_b2g`
+2. Setup your environment to fetch the custom NDK: `export LOCAL_NDK_BASE_URL='ftp://ftp.kaiostech.com/ndk/android-ndk'`
+3. Install Gecko dependencies: `cd gecko && ./mach bootstrap`, choose option 4 (Android Geckoview).
+4. Apply patch: `./patcher/patcher.sh`
```
Comment (lissyx) [inline README.md:72]: This is not here?

File: README.md [hunk 4]
```diff
@@ -64,6 +64,21 @@ At boot time, you might need `adb shell setenforce 0` for B2G to boot (flash_sar
...
+5. Build: `./build-gsi.sh [type name] systemimage`
+
+   type:
```
Comment (lissyx) [inline README.md:75]: How do I choose between AB or A ?

File: README.md [hunk 5]
```diff
@@ -64,6 +64,21 @@ At boot time, you might need `adb shell setenforce 0` for B2G to boot (flash_sar
...
+   - `gsi_arm64_ab` (arm64 partitionA/B)
```
Comment (lissyx) [inline README.md:76]: Fails with:
```
FAILED: ninja: unknown target 'gsi_arm64_ab'
```

File: README.md [hunk 6]
```diff
@@ -64,6 +64,21 @@ At boot time, you might need `adb shell setenforce 0` for B2G to boot (flash_sar
...
+   - `gsi_arm64_a` (arm64 partitionA)
```
Comment (lissyx) [inline README.md:77]:
```
FAILED: ninja: unknown target 'gsi_arm64_a'
```

File: README.md (hunk 4 above)
Comment (ittat) [reply README.md:75]: ab arm64
https://github.com/phhusson/treble_experimentations/wiki

File: README.md (hunk 6 above)
Comment (ittat) [reply README.md:77]: I need to check `build-gsi.sh`, there should be something wrong

File: README.md (hunk 3 above)
Comment (ittat) [reply README.md:72]: need it 
https://github.com/ittat/patcher/tree/b2g-gsi
https://github.com/ittat/manifests/blob/b2g-gsi/gsi.xml#L16

File: README.md (hunk 6 above)
Comment (ittat) [reply README.md:77]: did you try `build-gsi.sh gsi_arm64_ab systemimage` ?

Comment (lissyx) [reply README.md:77]: Maybe the pack of patching?

File: README.md (hunk 3 above)
Comment (lissyx) [reply README.md:72]: ok, it's not `+x`

Comment (lissyx) [reply README.md:72]: patching failure:
```
Application de  Add route to gateway on null netmask
error: le patch a échoué : src/java/com/android/internal/telephony/dataconnection/DataConnection.java:25
error: src/java/com/android/internal/telephony/dataconnection/DataConnection.java : le patch ne s'applique pas
le patch a échoué à 0001 Add route to gateway on null netmask
astuce: Utilisez 'git am --show-current-patch=diff' pour visualiser le patch en échec
```

File: README.md (hunk 6 above)
Comment (lissyx) [reply README.md:77]: yes ...
```
  PATH=$HOME/node_modules/.bin/:$PATH ./build-gsi.sh gsi_arm64_ab systemimage
```

Failing the same way after patching

File: README.md (hunk 3 above)
Comment (ittat) [reply README.md:72]: quite a bad day ~
We don’t actually need this patch...

**Comment by ittat** [comment]
@lissyx  I have fixed these problems and can do it normally in my side.
need `GITREPO=https://github.com/ittat/manifests BRANCH=b2g-gsi`
 `./patcher/patcher.sh` 
`./build-gsi.sh gsi_arm64_ab systemimage`

**Pull request closed without merging**